Automattic/wp-calypso
# Hosting Panel: Add selector and banner for Business plan upgrade

### Summary

This PR adds an upgrade banner to the top of the Hosting Configuration section if the site is not on a Business plan, in preparation to expose visibility of the section to Simple sites.

This work will be encapsulated behind a feature flag when #38155 is merged and the visibility selector for the Hosting section performs a feature flag check for `hosting/all-sites`.

<hr>

### Screenshots

**Upgrade banner displayed on Hosting section**

<img width="1159" alt="Screen Shot 2019-12-04 at 11 15 49 PM" src="https://user-images.githubusercontent.com/8892849/70203764-0cbc9580-16ec-11ea-95ef-c0e5e06801c5.png">

**Additional button added to Thank You page after upgrade**
<img width="991" alt="Screen Shot 2019-12-09 at 4 41 24 PM" src="https://user-images.githubusercontent.com/8892849/70475587-d6439980-1aa2-11ea-8fbe-ca9d50898a56.png">

<hr>

### Testing

* Start up Calypso with the `hosting/all-sites` flag running:
`ENABLE_FEATURES=hosting/all-sites npm start`
* **Banner display:**
    * Go to a Simple Site. Ensure that the 'Upgrade' banner is visible.
    * Click on the banner and ensure that you are directed to the shopping cart with the business plan added
    * Go to a site on a Business plan and ensure that the banner is not visible.
* **Thank you page button:**
    * Go to a Simple Site. Click on the Upgrade banner to upgrade your site.
    * After checking out, ensure that a second button is displayed that says "Return to Hosting" and that this button returns you to the Hosting Configuration panel for the site.
    * Test the non-hosting upgrade flow by copying the Thank You page URL, navigating to the Plugins section (or Themes section) of your site, then navigating to the Thank You page URL from your previous checkout. Ensure that the "Back to Hosting" button does not appear.

Fixes #36488
